core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java

```diff
@@ -1034,10 +1036,18 @@ public void addListener(Object listener, KeyFilter<? super K> filter, ClassLoade
       addListenerInternal(listener, DataConversion.IDENTITY_KEY, DataConversion.IDENTITY_VALUE, filter, converter, classLoader, false);
    }
 
+   /**
+    * Gets a suitable indexing provider for the given indexed filter.
+    *
+    * @param indexedFilter the filter
+    * @return the FilterIndexingServiceProvider that supports the given IndexedFilter or {@code null} if none was found
+    */
    private FilterIndexingServiceProvider findIndexingServiceProvider(IndexedFilter indexedFilter) {
-      for (FilterIndexingServiceProvider provider : filterIndexingServiceProviders) {
-         if (provider.supportsFilter(indexedFilter)) {
-            return provider;
+      if (filterIndexingServiceProviders != null) {
```

Comment:
I suppose it's possible that one thread is adding a listener just as another thread is stopping the cache, so the null check is still useful here.

Reply:
yup